buildSrc/src/main/kotlin/org/gradle/plugins/ideconfiguration/IdeConfigurationPlugin.kt

```diff
@@ -0,0 +1,448 @@
+                  <option name="SERVER_MODE" value="false" />
+                  <option name="SHMEM_ADDRESS" value="javadebug" />
+                  <option name="HOST" value="localhost" />
+                  <option name="PORT" value="5005" />
+                  <method />
+                </configuration>
+            """)
+    }
+
+    private fun Project.configureJunitRunConfiguration(runManagerComponent: org.jsoup.nodes.Element) {
+        val junitConfiguration = runManagerComponent.select("configuration[type=JUnit]").first()
+        val junitVmParametersOption = junitConfiguration.select("option[name=VM_PARAMETERS]").first()
+        junitVmParametersOption.attr("value", getDefaultJunitVmParameter(project("docs")))
+        val lang = System.getenv("LANG") ?: "en_US.UTF-8"
+        val envs = junitConfiguration.select("envs").first()
+        envs.createOrEmptyOutChildElement("env").attr("name", "LANG").attr("value", lang)
+    }
+
+    private fun configureSourceFolders(document: Document) {
+        val sourceFolders = document.select("component[name=NewModuleRootManager]").first().select("content").first().select("sourceFolder[url$=/resources]")
+
+        sourceFolders.forEach {
+            if (it.hasAttr("isTestSource")) {
+                it.removeAttr("isTestSource")
+                it.attr("type", "java-test-resource")
+            } else {
+                it.attr("type", "java-resource")
+            }
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun Project.configureLanguageLevel(ideaModule: IdeaModule) {
+        val ideaLanguageLevel = if ((findProperty("projectsRequiringJava8") as List<Project>).contains(ideaModule.project)) "1.8" else "1.6"
+        // Force everything to Java 6, pending detangling some int test cycles or switching to project-per-source-set mapping
+        ideaModule.languageLevel = IdeaLanguageLevel(ideaLanguageLevel)
+        ideaModule.targetBytecodeVersion = JavaVersion.toVersion(ideaLanguageLevel)
+    }
+
+    private fun removeGradleBuildOutputDirectories(module: Module) {
+        module.dependencies.removeAll {
+            it is ModuleLibrary &&
+                it.classes.any {
+                    // remove all Gradle build output directories from all scopes
+                    it.url.contains("/build/classes/") ||
+                        it.url.contains("/build/resources/") ||
+                        // remove possible resource directories from all scopes
+                        // replaced by java-resource/java-test-resource
+                        it.url.contains("${'$'}MODULE_DIR$/src/") ||
+                        // keep for build/generated-resources/main/*-classpath.properties
+                        // required by DefaultModuleRegistry
+                        it.url.contains("${'$'}MODULE_DIR$/build/") && !it.url.contains("generated-resources")
+                }
+        }
+    }
+
+    private fun Project.getRootExcludeDirs(): Set<File> {
```

Comment:
:o: could be a `val`:
```kotlin
private
val Project.rootExcludeDirs
    get() = setOf(
        file("intTestHomeDir"),
        file("buildSrc/build"),
        file("buildSrc/.gradle"))
```
And use site can use `rootExcludeDirs` instead of `getRootExcludeDirs()`